anydex/wallet/provider.py

```diff
@@ -0,0 +1,67 @@
+import abc
+
+
+class Provider(metaclass=abc.ABCMeta):
+    """
+    Abstract class to create providers from.
+    A provider is an abstraction used to interact with the blockchain.
+
+    For each cryptocurrency, a new abstract class specific to that
+    cryptocurrency should be created and any method specific to the cryptocurrency should be specified.
+    That class can than be subclassed and implemented.
+    """
+
+    @abc.abstractmethod
+    def submit_transaction(self, tx):
+        """
+        Submit a signed transaction to the network
+        :param tx: the signed transaction to submit to the network
+        :return: the transaction hash
+        """
+        return
+
+    @abc.abstractmethod
+    def get_balance(self, address):
```

Comment:
Is getting the balance of a wallet not a matter of offline database storage, rather than a provider one? In other words, should the `get_balance` function implemented in `Wallet` not be sufficient for our needs?

Reply:
No you would still need the block chain to get the latest information. You could also get all the transactions and save them in the database and then use those transactions to calculate the balance.